sdk/src/org.graalvm.nativeimage/src/org/graalvm/nativeimage/impl/ReflectionRegistry.java

```diff
@@ -46,7 +46,12 @@
 
 public interface ReflectionRegistry {
     default void register(ConfigurationCondition condition, Class<?>... classes) {
```

Comment:
Should we also throw a NPE if `condition` is null?

Reply:
That's already handled in https://github.com/oracle/graal/pull/6985/files#diff-cfb00f555d030e4938d0ee92feb053eb8ce70f3c182b36b80c0241bdd412e8a1R41-R43

Reply:
I'm not sure I see it. The first check seems `ConfigurationCondition.alwaysTrue().equals(condition)`, which is save but doesn't guarantee `condition` is not-null, no?

Reply:
Look at the highlighted part above the line you mention (the comment block makes things a bit harder).

![image](https://github.com/oracle/graal/assets/1435395/fdffd73a-a7a7-41f2-b375-d6e6b0f26802)

Reply:
Oh.... Sorry that's not testing the condition :facepalm:

Reply:
yes, that is just testing the runnable.

Reply:
Added https://github.com/oracle/graal/pull/6985/files#diff-cfb00f555d030e4938d0ee92feb053eb8ce70f3c182b36b80c0241bdd412e8a1R41-R44